scripts/funcTests/runFuncTests.sh

```diff
@@ -32,40 +36,59 @@ chmod +x scripts/funcTests/dotnet-install.sh
 
 # Get recommended version for bootstrapping testing version
 # Issue 8936 - DISABLED TEMPORARILY cli/dotnet-install.sh -i cli -c 2.2
-scripts/funcTests/dotnet-install.sh -i cli -c 2.2
+scripts/funcTests/dotnet-install.sh -i cli -c 2.2 -NoPath
+# cli/dotnet-install.sh -runtime dotnet -Channel 2.2 -i cli -NoPath
 
 DOTNET="$(pwd)/cli/dotnet"
 
-echo "initial dotnet cli install finished at `date -u +"%Y-%m-%dT%H:%M:%S"`"
 
-#restore solution packages
-$DOTNET msbuild -t:restore "$DIR/build/bootstrap.proj"
-if [ $? -ne 0 ]; then
-	echo "Restore failed!!"
-	exit 1
-fi
-
-echo "bootstrap project restore finished at `date -u +"%Y-%m-%dT%H:%M:%S"`"
+echo "dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
+
+# run it twice so dotnet cli can expand and decompress without affecting the result of the target
+dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting
+DOTNET_BRANCHES="$(dotnet msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting)"
+echo $DOTNET_BRANCHES | tr ";" "\n" |  while read -r DOTNET_BRANCH
+do
+	echo $DOTNET_BRANCH
+	ChannelAndVersion=($DOTNET_BRANCH)
+	Channel=${ChannelAndVersion[0]}
+	if [ ${#ChannelAndVersion[@]} -eq 1 ]
+	then
+		Version="latest"
+	else
+		Version=${ChannelAndVersion[1]}
+	fi
+	echo "Channel is: $Channel"
+	echo "Version is: $Version"
+	scripts/funcTests/dotnet-install.sh -i cli -c $Channel -v $Version -nopath
+
+	# Display current version
+	$DOTNET --version
+	dotnet --info
+done
 
-echo "$DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting"
-DOTNET_BRANCH="$($DOTNET msbuild build/config.props /v:m /nologo /t:GetCliBranchForTesting)"
+echo "initial dotnet cli install finished at `date -u +"%Y-%m-%dT%H:%M:%S"`"
 
-echo $DOTNET_BRANCH
-# Issue 8936 - TEMPORARILY using direct path to script
-scripts/funcTests/dotnet-install.sh -i cli -c $DOTNET_BRANCH
+echo "================="
 
-# Install the 2.x runtime because our tests target netcoreapp2x
-# Issue 8936 - TEMPORARILY using direct path to script
-scripts/funcTests/dotnet-install.sh -runtime dotnet -Channel 2.2 -i cli -NoPath
-# Display current version
-$DOTNET --version
+# install SDK2 runtime as we encounter problems on running dotnet vstest command when only download SDK3.
+cli/dotnet-install.sh -runtime dotnet -Channel 2.2 -i cli -NoPath
 
 echo "Deleting .NET Core temporary files"
 rm -rf "/tmp/"dotnet.*
 
 echo "second dotnet cli install finished at `date -u +"%Y-%m-%dT%H:%M:%S"`"
 echo "================="
 
+#restore solution packages
+$DOTNET msbuild -t:restore "$DIR/build/bootstrap.proj"
```

Comment:
I think this is reasonable.
The dotnet version installed in runFuncTests are for testing. It's usually the latest version of the specified channel. Since it's installed with -NoPath, we can only use $DOTNET to refer to it. 
If using `dotnet`, it's referring to the one for build. It's installed in "Install .NET 5.0 for build" step in yaml file, usually the stable version of the specified channel. Since it's added to PATH. We can use `dotnet `to refer to it.
So:
```
	# Display current version
	$DOTNET --version
```
This is displaying the dotnet we installed, so use $DOTNET to see the version
```
# restore packages
dotnet msbuild build/build.proj /t:Restore
```
we would like to use the stable version to run restore. (we used to have problem when using the latest version to restore, but not sure if we still have problem for now)
```
# Unit tests
dotnet msbuild build/build.proj /t:CoreUnitTests
```
```
# Func tests
dotnet msbuild build/build.proj /t:CoreFuncTests /p:VisualStudioVersion=16.0 /p:Configuration=Release /p:BuildNumber=1 /p:ReleaseLabel=beta
```
As the target CoreUnitTests and CoreFuncTests will first run build on test projects, and we would like to build them using stable version,so we use `dotnet`.
When the testing dlls are generated, running those dlls, the dotnet matters is [here](https://github.com/NuGet/NuGet.Client/blob/dev-feature-xplatVerification/build/build.proj#L389), and it's value is https://github.com/NuGet/NuGet.Client/blob/dev/build/common.project.props#L24-L25

It's complicated and not intuitive, I should add some comments on it, if you agree with the usage of dotnet and $DOTNET

Reply:
Line 84 of `runFuncTests.sh` is restoring "bootstrap" files needed for build. It is not running tests. As you wrote, the CoreUnitTests and CoreFuncTests targets need to find and use the dotnet cli installed in the `cli` folder. But `runFuncTests.sh` never runs tests directly and it's less confusing for it to use the same dotnet cli as the rest of the script.

Reply:
You're right. Changed to dotnet when restore bootstrap.proj. Thanks!